### PR TITLE
Fem: Remove unused view properties

### PR DESCRIPTION
**Pull request by marioalexis84**
To be merged after #13274.
Removes unused constraint symbol view properties.
Now the color of the symbol is controlled by the ShapeAppearance property.

### CODE REVIEW

File: src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp [hunk 1]
```diff
@@ -115,6 +123,32 @@ void ViewProviderFemConstraint::attach(App::DocumentObject* pcObject)
     addDisplayMaskMode(sep, "Base");
 }
 
+std::string ViewProviderFemConstraint::resourceSymbolDir =
+    App::Application::getResourceDir() + "Mod/Fem/Resources/symbols/";
+
+void ViewProviderFemConstraint::loadSymbol(const char* fileName)
+{
+    ivFile = fileName;
+    SoInput* in = new SoInput;
```
Comment (wwmayer) [inline src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp:115]: There is no need to create the SoInput on the heap. Just create it on the stack and pass it as `&in` to readAll().

File: src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp [hunk 2]
```diff
@@ -115,6 +123,32 @@ void ViewProviderFemConstraint::attach(App::DocumentObject* pcObject)
...
+    in->openFile(ivFile);
+    SoSeparator* nodes = SoDB::readAll(in);
```
Comment (wwmayer) [inline src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp:117]: If for some reason something goes wrong with the loading then `nodes` will be a null pointer. This should be checked before de-referencing it.

**Comment by marioalexis84** [comment]
@wwmayer  Is it planned to remove the Transparency property from Gui::ViewProviderGeometryObject?

File: src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp (hunk 2 above)
Comment (marioalexis84) [reply src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp]: Done

File: src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp (hunk 1 above)
Comment (marioalexis84) [reply src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp]: Done

**Comment by wwmayer** [comment]
> Is it planned to remove the Transparency property from Gui::ViewProviderGeometryObject?

That's a good question! Removing it would simplify quite a few things in the code but I am not sure if people like it to have the transparency as a child item of the Shape Appearance item.

The ideal solution would be to make the Transparency item a sibling item of the Shape Appearance item.

**Comment by yorikvanhavre** [comment]
@davesrocketshop could you summarize somewhere what would be needed for other workbenches to port to the new Material system?

**Comment by davesrocketshop** [comment]
> @davesrocketshop could you summarize somewhere what would be needed for other workbenches to port to the new Material system?

Starting a thread. There's a lot of stuff to cover so I'll add a post at a time

**Comment by davesrocketshop** [comment]
https://forum.freecad.org/viewtopic.php?t=87250
